lib/urbanopt/reopt/reopt_ghp_post_processor.rb

```diff
@@ -0,0 +1,155 @@
+# *********************************************************************************
+# URBANopt (tm), Copyright (c) Alliance for Sustainable Energy, LLC.
+# See also https://github.com/urbanopt/urbanopt-reopt-gem/blob/develop/LICENSE.md
+# *********************************************************************************
+
+require 'bundler/setup'
+require 'urbanopt/reopt/reopt_logger'
+require 'urbanopt/reopt/reopt_ghp_api'
+require 'csv'
+require 'json'
+require 'fileutils'
+
+module URBANopt # :nodoc:
+  module REopt # :nodoc:
+    class REoptGHPPostProcessor
+
+      def initialize(run_dir, system_parameter, modelica_result, reopt_ghp_assumptions = nil, nrel_developer_key = nil, localhost)
+        # initialize @@logger
+        @@logger ||= URBANopt::REopt.reopt_logger
+
+        @nrel_developer_key = nrel_developer_key
+        @localhost = localhost
+        @reopt_ghp_output_district = nil
+        @reopt_ghp_output_building = []
+        @reopt_ghp_assumptions_hash = nil
+        @reopt_ghp_assumptions = nil
+        @system_parameter = nil
+        @system_parameter_hash = nil
+        @modelica_result = nil
+        @building_ids = nil
+        @run_dir = run_dir
+
+        if !reopt_ghp_assumptions.nil?
+          @reopt_ghp_assumptions = reopt_ghp_assumptions
+          File.open(reopt_ghp_assumptions, 'r') do |file|
+            @reopt_ghp_assumptions_input_hash = JSON.parse(file.read, symbolize_names: true)
+          end
+        end
+
+        if !system_parameter.nil?
+          @system_parameter = system_parameter
+          File.open(system_parameter, 'r') do |file|
+            @system_parameter_input_hash = JSON.parse(file.read, symbolize_names: true)
+          end
+          #Determine loop order
+          loop_order = File.join(File.dirname(system_parameter), '_loop_order.json')
+          if File.exist?(loop_order)
+            File.open(loop_order, 'r') do |file|
+              loop_order_input= JSON.parse(file.read, symbolize_names: true)
+              # Check the type of the parsed data
+              if loop_order_input.is_a?(Array)
+                @loop_order_input_hash = loop_order_input
+                @loop_order_input_hash.each do |item|
+                  puts "Building IDs in group: #{item[:list_bldg_ids_in_group].inspect}"
+                  puts "GHE IDs in group: #{item[:list_ghe_ids_in_group].inspect}"
+                end
+              elsif loop_order_input.is_a?(Hash)
```

Comment:
I think it's safe to assume loop_order will always be a list from ThermalNetwork. On that note, perhaps it should not be named `loop_order_input_hash`, is that true?

Reply:
i was running into errors which is why i have added extra checks here